text/0502_aws-vpclattice.md

```diff
@@ -0,0 +1,1031 @@
+      accessMode: vpclattice.RuleAccessMode.UNAUTHENTICATED
+    });
+
+    //add a listenerRule that will use the goodbyeworld lambda as a Target
+    mylistener.addListenerRule({
+      name: 'goodbyeworld',
+      priority: 20,
+      action: [
+        {
+          targetGroup: new vpclattice.TargetGroup(this, 'goodbyelambdatargets', {
+            name: 'goodbyeworld',
+            target: vpclattice.Target.lambda([
+              support.goodbyeWorld,
+            ]),
+          }),
+        },
+      ],
+      
+      httpMatch: {
+        pathMatches: { path: '/goodbye' },
+      },
+      // we will only allow access to this service from the ec2 instance
+      allowedPrincipals: [support.ec2instance.role],
+      accessMode: vpclattice.RuleAccessMode.AUTHENTICATED_ONLY,
+    });
+
+    
+    myLatticeService.applyAuthPolicy();
+
+    /**
+     * Create a ServiceNetwork.
+     * OPINIONATED DEFAULT: The default behavior is to create a
+     * service network that requries an IAM policy, and authenticated access
+     * ( requestors must send signed requests )
+     */
+
+    const serviceNetwork = new ServiceNetwork(this, 'LatticeServiceNetwork', {
+      services: [myLatticeService],
+      vpcs: [
+        support.vpc1,
+      ],
+    });
+
+    serviceNetwork.applyAuthPolicyToServiceNetwork();
+  }
+}
+```
+
+---
+
+## Proposed API Design for vpclattice
+
+[(The Api is additionally documented on Constructs.dev here)](https://constructs.dev/packages/aws-vpclattice-prealpha/v/0.4.11/api/AuthType?lang=typescript)
+
+### Constructs
+
+- ✓ [AssociateVpc](#associatevpc)
+- ✓ [ServiceNetwork](#servicenetwork)
+- ✓ [ServiceNetworkAssociation] (#servicenetworkassociation)
+- ✓ [Service](service)
+- ✓ [ServiceAssociation]
+- ✓ [Listener](#listener)
+- ✓ [TargetGroup](#targetgroup)
+
+### Classes
+
+- [HealthCheck]
+- [LoggingDestination]
+- [Target]
+
+### Enums
+
+- [AuthType]
+- [FixedResponse]
+- [HTTPMethods]
+- [IpAddressType]
+- [MatchOperator]
+- [PathMatchType]
+- [Protocol]
+- [ProtocolVersion]
+- [RuleAccessMode]
+- [ServiceNetworkAccessMode]
+- [TargetType]
+
+---
+
+## Constructs Details
+
+### AssociateVpc
+
+Creates a CloudFormation `AWS::VpcLattice::ServiceNetworkVpcAssociation`
+
+This construct associates a VPC, with a service network.  It creates an endpoint interface in a VPC to enable the resources in the vpc
+to be clients of the service network.
+
+The construct extends core.Resource, and will consume AssociateVpcProps.
+
+```typescript
+/**
+ * Props to Associate a VPC with a Service Network
+ */
+export interface AssociateVpcProps {
+  /**
+   * security groups for the lattice endpoint
+   * @default a security group that will permit inbound 443
+   */
+  readonly securityGroups?: ec2.ISecurityGroup[];
+  /**
+   * The VPC to associate with
+   */
+  readonly vpc: ec2.IVpc;
+  /**
+   * Service Network Identifier
+   */
+  readonly serviceNetworkId: string;
```

Comment:
```suggestion
  readonly serviceNetwork: IServiceNetwork;
```

What about this instead?

Reply:
The AssociateVPC class should have been private, and not exported.     The association of the service network to the vpc, is done via the `associateVPC` method on the ServiceNetwork.